src/ESLint.Formatter/sarif.js

```diff
@@ -178,8 +207,17 @@ module.exports = function (results, data) {
         });
     }
 
-    if (sarifResults.length > 0) {
-        sarifLog.runs[0].results = sarifResults;
+    // Per the SARIF spec §3.14.23, run.results must be present even if there are no results.
+    // This provides a positive indication that the run completed and no results were found.
+    sarifLog.runs[0].results = sarifResults;
```

Comment:
it is acceptable for run.results to be null in cases where a catastrophic engine error prevented analysis from occurring. are you permitting this case?

e.g., if a user passes a single syntactically invalid file to analyze, run.results may be null (in cases where a scanner cannot function against syntactically incorrect code).

Reply:
"If the tool started but failed to begin its analysis (for example, because its command line was invalid), then again results MAY be present, and if present SHALL be null."

---
In reply to: [385957192](https://github.com/microsoft/sarif-sdk/pull/1793#discussion_r385957192) [](ancestors = 385957192)

Reply:
As we discussed offline: the engine did not fail to begin its analysis; it can produce real results before (and after) hitting this problem.

---
In reply to: [385958991](https://github.com/microsoft/sarif-sdk/pull/1793#discussion_r385958991) [](ancestors = 385958991,385957192)